### PR TITLE
examples: Add an example for doing debug

### PR DESCRIPTION
**Pull request by larry-safran**
Added for fixit 

Fixes b/259286633

### CODE REVIEW

File: examples/build.gradle [hunk 1]
```diff
@@ -43,7 +43,7 @@ dependencies {
 protobuf {
     protoc { artifact = "com.google.protobuf:protoc:${protocVersion}" }
     plugins {
-        grpc { artifact = "io.grpc:protoc-gen-grpc-java:${grpcVersion}" }
+        grpc { artifact = "io.grpc:protoc-gen-grpc-java:1.54.0" }
```
Comment (temawi) [inline examples/build.gradle:46]: You probably didn't mean to leave this change in?

File: examples/pom.xml [hunk 2]
```diff
@@ -97,7 +97,7 @@
         <configuration>
           <protocArtifact>com.google.protobuf:protoc:${protoc.version}:exe:${os.detected.classifier}</protocArtifact>
           <pluginId>grpc-java</pluginId>
-          <pluginArtifact>io.grpc:protoc-gen-grpc-java:${grpc.version}:exe:${os.detected.classifier}</pluginArtifact>
+          <pluginArtifact>io.grpc:protoc-gen-grpc-java:1.54.0:exe:${os.detected.classifier}</pluginArtifact>
```
Comment (temawi) [inline examples/pom.xml:100]: Revert this?

File: examples/example-debug/src/main/java/io/grpc/examples/debug/HelloWorldDebuggableClient.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    if (args.length > 1) {
+      target = args[1];
+      //
+      // Parse port number from target and add 1000 to it for debug port
+      String[] split = target.split(":");
+      if (split.length == 2) {
+        try {
+          debugPort = Integer.parseInt(split[1]) + 1000;
+        } catch (NumberFormatException e) {
+            System.err.println("Invalid port number " + split[1]);
```
Comment (temawi) [inline examples/example-debug/src/main/java/io/grpc/examples/debug/HelloWorldDebuggableClient.java:98]: nit: The indent is off.

**Review by temawi — APPROVED** [review]
Approved, but take a look at the few comments I left.

File: examples/build.gradle (hunk 1 above)
Comment (larry-safran) [reply examples/build.gradle]: Undid the change

File: examples/pom.xml (hunk 2 above)
Comment (larry-safran) [reply examples/pom.xml]: Done

**Pull request merged**